durablefunctionsmonitor.dotnetisolated.core/Common/Auth.cs

```diff
@@ -334,6 +324,48 @@ private static string TryGetHubNameFromHostJson()
             }
         }
 
+        private static async Task<ClaimsPrincipal> GetClaimsPrincipal(HttpRequestData request, string userNameClaimName, string rolesClaimName)
+        {
+            // First trying the request object
+            var principal = new ClaimsPrincipal(request.Identities);
+
+            if (principal.Identity != null && principal.Identity.IsAuthenticated && principal.HasClaim(c => c.Type == userNameClaimName))
+            {
+                return principal;
+            }
+
+            // Then trying to parse the x-ms-client-principal header
+            if (request.Headers.TryGetValues("x-ms-client-principal", out var clientPrincipalHeaderValues))
+            {
+                string clientPrincipalJson = Encoding.UTF8.GetString(Convert.FromBase64String(clientPrincipalHeaderValues.Single()));
+
+                dynamic clientPrincipal = JObject.Parse(clientPrincipalJson);
+
+                var identity = new ClaimsIdentity((string)clientPrincipal.auth_typ);
+                var claims = (IEnumerable<dynamic>)clientPrincipal.claims;
+
+                var userNameClaim = claims.SingleOrDefault(c => c.typ == userNameClaimName);
+                if (userNameClaim != null)
+                {
+                    string userNameClaimValue = userNameClaim.val;
+
+                    identity.AddClaim(new Claim(userNameClaimName, userNameClaimValue));
+                }
+
+                var roles = claims.Where(c => c.typ == rolesClaimName).Select(c => (string)c.val);
+                identity.AddClaims(roles.Select(r => new Claim(rolesClaimName, r)));
+
+                principal = new ClaimsPrincipal(identity);
+                return principal;
```

Comment:
This makes sense, though I'm definitely a bit surprised that we have to do this, I would have hoped this would "just work". Let's see if we can get confirmation from @cgillum or someone with auth expertise to confirm we're not missing something.

Reply:
extracting from this header does appear to be the recommendation from MS: https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-http-webhook-trigger?tabs=python-v2%2Cisolated-process%2Cnodejs-v4%2Cfunctionsv2&pivots=programming-language-csharp#working-with-client-identities

Reply:
That was precisely the point discovered in #161 - the runtime does give us the ClientPrincipal, when we're a plain Isolated func, but does not give it to us, when we're under ASP.Net Core Integration (while still being a plain Isolated func).
Yes, I do believe it is worth fixing this at the runtime level, but until then here's basically a workaround.